assets/app/view/game/buy_trains.rb

```diff
@@ -288,6 +303,30 @@ def remaining_trains
           *rows,
         ])
       end
+
```

Comment:
Add buttons above the train buy elements to select which shell the player is buying for and store within the page. The train buying view will use this information to decide whether or not a train is buyable in that shell and if the president can contribute funds.